streams.cpp

```diff
@@ -40,7 +40,11 @@ bool DS::Stream::readLine(void* buffer, size_t count)
             break;
     }
     *outp = 0;
-    return !eof;
+    // If the last line in the file doesn't end in a newline,
+    // the loop above may have reached EOF, but still have read some characters.
+    // In that case, we shouldn't signal EOF to the caller yet,
+    // so that the characters from the last line are read normally.
+    return !eof || outp != buffer;
```

Comment:
It seems we don't even need the `!eof` check here...  Since newlines are stored in the output buffer, we will always write at least one character except in the EOF case.